File: app/src/main/java/com/karkoszka/cookingtime/services/CTBroadcastReceiver.java
package com.karkoszka.cookingtime.services;

import android.content.Context;
import android.content.Intent;
import android.os.SystemClock;
import androidx.legacy.content.WakefulBroadcastReceiver;
import android.util.Log;

import com.karkoszka.cookingtime.activities.MainActivity;

public class CTBroadcastReceiver extends WakefulBroadcastReceiver {
	
	
	@Override
	  public void onReceive(Context context, Intent intent) {
		int plate = intent.getIntExtra(MainActivity.ALARM_OFF_PLATE_NO, 0);
	    Intent intent2 = new Intent(context, AlarmSoundService.class);
	    intent2.putExtra(MainActivity.ALARM_OFF_PLATE_NO, plate);
		intent2.putExtra(AlarmSoundService.START_PLAY, true);

        Log.i("SimpleWakefulReceiver", "Starting service @ " + SystemClock.elapsedRealtime());
        startWakefulService(context, intent2);

	  }
	
}

File: app/src/main/java/com/karkoszka/cookingtime/services/CTBroadcastReceiver.kt
package com.karkoszka.cookingtime.services

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.SystemClock
import android.util.Log
import com.karkoszka.cookingtime.activities.MainActivity

class CTBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val plate = intent.getIntExtra(MainActivity.ALARM_OFF_PLATE_NO, 0)
        val intent2 = Intent(context, AlarmSoundService::class.java)
        intent2.putExtra(MainActivity.ALARM_OFF_PLATE_NO, plate)
        intent2.putExtra(AlarmSoundService.START_PLAY, true)
        Log.i("SimpleWakefulReceiver", "Starting service @ " + SystemClock.elapsedRealtime())
        context.startService(intent2)
    }
}